### PR TITLE
Split MergeTrafficPolicy and make it more readable and easy to maintain

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@howardjohn @ramaraochavali

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -758,59 +758,92 @@ func MaybeBuildStatefulSessionFilterConfig(svc *model.Service) *statefulsession.
 	return nil
 }
 
-// MergeTrafficPolicy returns the merged TrafficPolicy for a destination-level and subset-level policy on a given port.
-func MergeTrafficPolicy(original, subsetPolicy *networking.TrafficPolicy, port *model.Port) *networking.TrafficPolicy {
-	if subsetPolicy == nil {
-		return original
+// GetPortLevelTrafficPolicy return the port level traffic policy
```
Comment (ramaraochavali) [inline pilot/pkg/networking/util/util.go:761]: Can you add a comment on what is the bool return value for?

File: pilot/pkg/networking/util/util.go [hunk 2]
```diff
@@ -758,59 +758,92 @@ func MaybeBuildStatefulSessionFilterConfig(svc *model.Service) *statefulsession.
...
-				// per the docs, port level policies do not inherit and instead to defaults if not provided
-				mergedPolicy.ConnectionPool = p.ConnectionPool
-				mergedPolicy.OutlierDetection = p.OutlierDetection
-				mergedPolicy.LoadBalancer = p.LoadBalancer
-				mergedPolicy.Tls = p.Tls
-				break
-			}
-		}
+// Shallowcopy a traffic policy, portLevelSettings are ignorred.
+func CloneTrafficPolicy(original *networking.TrafficPolicy) *networking.TrafficPolicy {
```
Comment (ramaraochavali) [inline pilot/pkg/networking/util/util.go:837]: Can you add a fuzz test for this to ensure the fields are not missed in future?

File: pilot/pkg/networking/util/util.go [hunk 3]
```diff
@@ -758,59 +758,92 @@ func MaybeBuildStatefulSessionFilterConfig(svc *model.Service) *statefulsession.
...
-			if p.Port != nil && uint32(port.Port) == p.Port.Number {
-				// per the docs, port level policies do not inherit and instead to defaults if not provided
-				mergedPolicy.ConnectionPool = p.ConnectionPool
-				mergedPolicy.OutlierDetection = p.OutlierDetection
-				mergedPolicy.LoadBalancer = p.LoadBalancer
-				mergedPolicy.Tls = p.Tls
-				break
-			}
-		}
+// Shallowcopy a traffic policy, portLevelSettings are ignorred.
```
Comment (ramaraochavali) [inline pilot/pkg/networking/util/util.go:836]: nit: /s/ignorred/ignored

File: pilot/pkg/networking/util/util.go (hunk 2 above)
Comment (howardjohn) [inline pilot/pkg/networking/util/util.go:837]: nit: can we call it ShallowCopyTrafficPolicy? Usually Clone implies deep copy

Comment (hzxuzhonghu) [reply pilot/pkg/networking/util/util.go:837]: It is not copying call fields

File: pilot/pkg/networking/util/util.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/util/util.go:761]: sure

File: pilot/pkg/networking/util/util.go (hunk 3 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/util/util.go:836]: ack

**Comment by hzxuzhonghu** [comment]
addressed all your comments

File: pilot/pkg/networking/util/util.go (hunk 3 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/util/util.go:835]: this is pending still

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**